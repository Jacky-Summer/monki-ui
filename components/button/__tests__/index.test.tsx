import React from 'react'
import { shallow, render } from 'enzyme'
import Button, { ButtonProps } from '../button'

const testProps: ButtonProps = {
  type: 'primary',
  size: 'lg',
  className: 'monkey',
}

const disabledProps: ButtonProps = {
  disabled: true,
  onClick: jest.fn(),
}

describe('Button component', () => {
  it('should render Buttons of different type correctly', () => {
    const wrapper = render(
      <div>
        <Button>default</Button>
        <Button type="primary">primary</Button>
        <Button type="info">info</Button>
        <Button type="warning">warning</Button>
        <Button type="danger">danger</Button>
        <Button type="dashed">dashed</Button>
        <Button type="text">text</Button>
      </div>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a link when btnType equals link and href is provided', () => {
    const wrapper = shallow(
      <Button type="link" href="https://github.com/Jacky-Summer">
        Link
      </Button>,
    )
    const element = wrapper.first()
    expect(element.name()).toEqual('a')
    expect(element.hasClass('mk-btn mk-btn-link'))
  })

  it('should render the correct component based on different props', () => {
    const wrapper = shallow(<Button {...testProps}>Test</Button>)
    const element = wrapper.find('.monkey')
    expect(element.hasClass('mk-btn-primary mk-btn-lg')).toBeTruthy()
  })

  it('should render Buttons of different size correctly', () => {
    const wrapper = render(
      <div>
        <Button type="primary" size="lg">
          lg尺寸
        </Button>
        <Button type="primary" size="md">
          md尺寸
        </Button>
        <Button type="primary" size="sm">
          sm尺寸
        </Button>
      </div>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render disabled button when disabled set to true', () => {
    const wrapper = shallow(<Button disabled>Disabled Button</Button>)
    const element = wrapper.find('button')
    expect(element.prop('disabled')).toBeTruthy()
    element.simulate('click')
    expect(disabledProps.onClick).not.toHaveBeenCalled()
  })

  it('should not render as link button when href is undefined', async () => {
    const wrapper = shallow(
      <Button type="link" href={undefined}>
        button
      </Button>,
    )
    expect(wrapper.find('a').exists()).toBeFalsy()
    expect(wrapper.find('button').exists()).toBeTruthy()
  })
})
