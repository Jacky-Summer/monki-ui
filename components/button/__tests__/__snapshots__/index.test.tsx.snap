// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render Buttons of different size correctly 1`] = `
<div>
  <button
    class="mk-btn mk-btn-primary mk-btn-lg"
    type="button"
  >
    <span>
      lg尺寸
    </span>
  </button>
  <button
    class="mk-btn mk-btn-primary mk-btn-md"
    type="button"
  >
    <span>
      md尺寸
    </span>
  </button>
  <button
    class="mk-btn mk-btn-primary mk-btn-sm"
    type="button"
  >
    <span>
      sm尺寸
    </span>
  </button>
</div>
`;

exports[`Button component should render Buttons of different type correctly 1`] = `
<div>
  <button
    class="mk-btn mk-btn-default mk-btn-md"
    type="button"
  >
    <span>
      default
    </span>
  </button>
  <button
    class="mk-btn mk-btn-primary mk-btn-md"
    type="button"
  >
    <span>
      primary
    </span>
  </button>
  <button
    class="mk-btn mk-btn-info mk-btn-md"
    type="button"
  >
    <span>
      info
    </span>
  </button>
  <button
    class="mk-btn mk-btn-warning mk-btn-md"
    type="button"
  >
    <span>
      warning
    </span>
  </button>
  <button
    class="mk-btn mk-btn-danger mk-btn-md"
    type="button"
  >
    <span>
      danger
    </span>
  </button>
  <button
    class="mk-btn mk-btn-dashed mk-btn-md"
    type="button"
  >
    <span>
      dashed
    </span>
  </button>
  <button
    class="mk-btn mk-btn-text mk-btn-md"
    type="button"
  >
    <span>
      text
    </span>
  </button>
</div>
`;
